utils/notify_issues.py

```diff
@@ -320,133 +412,191 @@ def notify_closed_issues(
                 base=github_baseurl,
                 prefix=github_repo_prefix,
-                repo=self.src_dir.name,
+                repo=self.source_dir.name,
             )
         git_log_url = "{git_url}/compare/{prev_commit}...{curr_commit}".format(
-            git_url=git_url, prev_commit=previous_commit, curr_commit=current_commit
+            git_url=git_url,
+            prev_commit=previous_commit,
+            curr_commit=current_commit,
         )
 
-        component = self.src_dir.name
+        component = self.source_dir.name
 
         for issue in closed_issues:
             log.info(f"Adding a comment to issue #{issue}")
             if message_template_path:
-                message = (
+                issue_message: Optional[str] = (
                     open(message_template_path, "r")
                     .read()
-                    .replace("@DIST@", self.dist.name)
-                    .replace("@PACKAGE_SET@", self.dist.package_set)
-                    .replace("@PACKAGE_NAME@", self.package_name)
+                    .replace("@DIST@", dist.name)
+                    .replace("@PACKAGE_SET@", dist.package_set)
+                    .replace("@PACKAGE_NAME@", package_name)
                     .replace("@COMPONENT@", component)
                     .replace("@REPOSITORY@", repo_type)
                     .replace("@RELEASE_NAME@", self.release_name)
                     .replace("@GIT_LOG@", shortlog)
                     .replace("@GIT_LOG_URL@", git_log_url)
                 )
             else:
-                message = None
+                issue_message = None
 
-            self.comment_issue(issue, message, add_labels, delete_labels)
+            self.comment_issue(issue, issue_message, add_labels, delete_labels)
 
-    def notify_build_report(
+    def run(
         self,
-        dist_label,
-        repo_type,
-        add_labels,
-        delete_labels,
-        commit_sha,
-        previous_stable_commit_sha,
+        command,
+        dist,
+        package_name,
+        repository_type=None,
+        repository_url=None,
         build_status=None,
+        state_file=None,
+        stable_state_file=None,
         build_log=None,
         additional_info=None,
-        repository_url=None,
     ):
+        if dist.package_set == "host":
+            dist_label = "host"
+        else:
+            dist_label = dist.distribution
+
+        current_commit = self.get_current_commit()
+        previous_stable_commit = None
+
+        if command == "upload" and repository_type == "current":
+            repository_type = "stable"
+
+        add_labels, delete_labels = self.get_labels(
+            command=command,
+            repository_type=repository_type,
+            build_status=build_status,
+            dist_label=dist_label,
+            package_name=package_name,
+        )
+
+        if command == "upload":
+            build_status = "uploaded"
```

Comment:
This is wrong, it will notify failed upload as successful one.

Reply:
Yes indeed I did not catched it in the refactor thank you. Let me fix that.